packages/expo-facebook/android/src/main/java/expo/modules/facebook/FacebookModule.kt

```diff
@@ -0,0 +1,342 @@
+    if (FacebookSdk.getApplicationId() == null) {
+      promise.reject(
+        ERR_FACEBOOK_MISCONFIGURED,
+        "No appId configured, required for initialization. " +
+          "Please ensure that you're either providing `appId` to `initializeAsync` as an argument or inside AndroidManifest.xml."
+      )
+    }
+
+    // Log out
+    AccessToken.setCurrentAccessToken(null)
+    LoginManager.getInstance().logOut()
+
+    // Convert permissions
+    val permissions = config.getList("permissions", listOf("public_profile", "email")) as List<String?>
+    if (config.containsKey("behavior")) {
+      var behavior = LoginBehavior.NATIVE_WITH_FALLBACK
+      when (config.getString("behavior")) {
+        "browser" -> behavior = LoginBehavior.WEB_ONLY
+        "web" -> behavior = LoginBehavior.WEB_VIEW_ONLY
+      }
+      LoginManager.getInstance().loginBehavior = behavior
+    }
+    LoginManager.getInstance().registerCallback(
+      callbackManager,
+      object : FacebookCallback<LoginResult> {
+        override fun onSuccess(loginResult: LoginResult) {
+          LoginManager.getInstance().registerCallback(callbackManager, null)
+
+          // This is the only route through which we send an access token back. Make sure the
+          // application ID is correct.
+          if (appId != loginResult.accessToken.applicationId) {
+            promise.reject(IllegalStateException("Logged into wrong app, try again?"))
+            return
+          }
+          val response = accessTokenToBundle(loginResult.accessToken)
+          response?.putString("type", "success")
+          promise.resolve(response)
+        }
+
+        override fun onCancel() {
+          LoginManager.getInstance().registerCallback(callbackManager, null)
+          promise.resolve(
+            Bundle().apply {
+              putString("type", "cancel")
+            }
+          )
+        }
+
+        override fun onError(error: FacebookException) {
+          LoginManager.getInstance().registerCallback(callbackManager, null)
+          promise.reject(ERR_FACEBOOK_LOGIN, "An error occurred while trying to log in to Facebook", error)
+        }
+      }
+    )
+    try {
+      val activityProvider: ActivityProvider by moduleRegistry()
+      LoginManager.getInstance().logInWithReadPermissions(activityProvider.currentActivity, permissions)
+    } catch (e: FacebookException) {
+      promise.reject(ERR_FACEBOOK_LOGIN, "An error occurred while trying to log in to Facebook", e)
+    }
+  }
+
+  @ExpoMethod
+  fun setFlushBehaviorAsync(flushBehavior: String, promise: Promise) {
+    AppEventsLogger.setFlushBehavior(AppEventsLogger.FlushBehavior.valueOf(flushBehavior.toUpperCase(Locale.ROOT)))
+    promise.resolve(null)
+  }
+
+  @ExpoMethod
+  fun logEventAsync(eventName: String?, valueToSum: Double, parameters: ReadableArguments?, promise: Promise) {
+    appEventLogger!!.logEvent(eventName, valueToSum, bundleWithNullValuesAsStrings(parameters))
```

Comment:
Can we reject instead of crashing when `appEventLogger` is null?

Reply:
Yes, I'll fix it